### PR TITLE
[stable/sonarqube] Update ingress apiVersion

### LINKED ISSUE #18999: [stable/sonarqube] Kubernetes v1.16 support
**Describe the bug**
Installing this chart in a 1.16 cluster causes an error due to the deprecation of the `Ingress` resource type in the `extensions/v1beta1` apiVersion. 

**Which chart**:
`stable/sonarqube`

### PR DESCRIPTION
**Pull request by oscarlofwenhamn**
#### What this PR does / why we need it:
This makes it possible to use the chart with an ingress in a v1.16 k8s cluster.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #18999 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Review by dieuph — APPROVED** [review]

**Comment by oscarlofwenhamn** [comment]
@rjkernick or @tsiddique, I think this requires another review or a
/ok-to-test
to proceed.

**Comment by Hokwang** [comment]
this is not good solution. 
please add k8s version check and select api logic.

File: stable/sonarqube/templates/ingress.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {{- if .Values.ingress.enabled -}}
 {{- $serviceName := include "sonarqube.fullname" . -}}
 {{- $servicePort := .Values.service.externalPort -}}
-apiVersion: extensions/v1beta1
+apiVersion: networking.k8s.io/v1beta1
```
Comment (rjkernick) [inline stable/sonarqube/templates/ingress.yaml:4]: I think you mean: "networking.k8s.io/v1"

File: stable/sonarqube/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: sonarqube
 description: Sonarqube is an open sourced code quality scanning tool
-version: 3.2.5
+version: 3.2.6
```
Comment (rjkernick) [inline stable/sonarqube/Chart.yaml]: Please bump the minor version instead, as this changes behavior.

**Review by rjkernick — CHANGES_REQUESTED** [review]

File: stable/sonarqube/templates/ingress.yaml (hunk 1 above)
Comment (davidkarlsen) [reply stable/sonarqube/templates/ingress.yaml:4]: also using https://helm.sh/docs/topics/chart_template_guide/builtin_objects/ may make it work in older and newer clusters.

**Comment by Hokwang** [comment]
please merge this. this is critical problem for who uses k8s 1.16+.

File: stable/sonarqube/templates/ingress.yaml (hunk 1 above)
Comment (usmonster) [reply stable/sonarqube/templates/ingress.yaml:4]: FYI, pending PR #19836 includes a fix that addresses @davidkarlsen's & @Hokwang's concern.

**Comment by oscarlofwenhamn** [comment]
Superseded by #19836. 
I've been out cold for the past month, hence no more work done. Big thanks to @usmonster for the improved PR.

**Pull request closed without merging**